src/Juvix/Compiler/Core/Data/IdentDependencyInfo.hs

```diff
@@ -53,3 +56,45 @@ createSymbolDependencyInfo tab = createDependencyInfo graph startVertices
 
 recursiveIdents :: InfoTable -> HashSet Symbol
 recursiveIdents = nodesOnCycles . createCallGraph
+
+-- | identifiers from which some recursive identifier can be reached
+recursiveIdentsClosure :: InfoTable -> HashSet Symbol
+recursiveIdentsClosure tab =
+  -- unfortunately, there is no Graph library function which would allow to
+  -- compute this in linear time; hence, we implement this directly
+  run $
+    evalState (mempty :: HashSet Symbol) $
+      foldM (dfs mempty) mempty (HashMap.keys graph)
```

Comment:
the formatted code looks nicer if we write vertical chains following the pattern `f . g . h $ x` instead of `f $ g $ h $ x`

```suggestion
  run .
    evalState (mempty :: HashSet Symbol) $
      foldM (dfs mempty) mempty (HashMap.keys graph)
```